doc/languages-frameworks/bower.section.md

```diff
@@ -0,0 +1,119 @@
+
+The resulting derivation is a union of all the downloaded Bower packages (and their dependencies). To use it, they still need to be linked together by Bower, which is where *buildBowerComponents* is useful.
+
+## `buildBowerComponents` function {#ssec-build-bower-components}
+
+The function is implemented in [pkgs/development/bower-modules/generic/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/bower-modules/generic/default.nix).
+
+### Example buildBowerComponents {#ex-buildBowerComponents}
+```nix
+bowerComponents = buildBowerComponents {
+  name = "my-web-app";
+  generated = ./bower-packages.nix; <co xml:id="ex-buildBowerComponents-1" />
+  src = myWebApp; <co xml:id="ex-buildBowerComponents-2" />
+};
+```
+
+In [Example 15.2, "buildBowerComponents"](#ex-buildBowerComponents) the following arguments are of special significance to the function:
+
+* `generated` specifies the file which was created by `bower2nix`.
+* `src` is your project's sources. It needs to contain a *bower.json* file.
+
+`buildBowerComponents` will run Bower to link together the output of `bower2nix`, resulting in a *bower_components* directory which can be used.
+
+Here is an example of a web frontend build process using `gulp`. You might use `grunt`, or anything else.
+
+### Example build script (*gulpfile.js*) {#ex-bowerGulpFile}
+```javascript
+var gulp = require('gulp');
+
+gulp.task('default', [], function () {
+  gulp.start('build');
+});
+
+gulp.task('build', [], function () {
+  console.log("Just a dummy gulp build");
+  gulp
+    .src(["./bower_components/**/*"])
+    .pipe(gulp.dest("./gulpdist/"));
+});
+```
+
+### Example Full example — *default.nix* {#ex-buildBowerComponentsDefaultNix}
+```nix
+{ myWebApp ? { outPath = ./.; name = "myWebApp"; }
+, pkgs ? import <nixpkgs> {}
+}:
+
+pkgs.stdenv.mkDerivation {
+  name = "my-web-app-frontend";
+  src = myWebApp;
+
+  buildInputs = [ pkgs.nodePackages.gulp ];
+
+  bowerComponents = pkgs.buildBowerComponents {
+    name = "my-web-app";
+    generated = ./bower-packages.nix;
+    src = myWebApp;
+  };
+
+  buildPhase = ''
+    cp --reflink=auto --no-preserve=mode -R $bowerComponents/bower_components .
+    export HOME=$PWD <co xml:id="ex-buildBowerComponentsDefault-3" />
+    ${pkgs.nodePackages.gulp}/bin/gulp build
+  '';
+
+  installPhase = "mv gulpdist $out";
+}
+```
+A few notes about [Example 15.4, Full example — *default.nix*](#ex-buildBowerComponentsDefaultNix):
+
+* The result of `buildBowerComponents` is an input to the frontend build.
+
+* Whether to symlink or copy the *bower_components* directory depends on the build tool in use. In this case a copy is used to avoid `gulp` silliness with permissions.
+
+* `gulp` requires `HOME` to refer to a writeable directory.
+
+* The actual build command. Other tools could be used.
+
+## Troubleshooting {#ssec-bower2nix-troubleshooting}
+
+`ENOCACHE` errors from `buildBowerComponents`
```

Comment:
This should probably be heading. And the stuff below, paragraphs instead of quotes.

Reply:
Updated.